### ISSUE TITLE
Regex Manager contains disallowed fields: commitMessagePrefix

### ISSUE THREAD

**Issue by micheelengronne** [renovatebot/config-help#617]
I would like to have a different commit prefix for updates from regex managers.

I tried:

```
  "regexManagers": [
    {
      "fileMatch": [".*molecule.yml$"],
      "matchStrings": [
        "image: (?<depName>.*?):(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)"
      ],
      "datasourceTemplate": "docker",
      "commitMessagePrefix": "test(renovate):"
    }
  ]
```
But I have the following message:
`Regex Manager contains disallowed fields: commitMessagePrefix`

Is there a way to set a different commitMessagePrefix for this regexManager ?

Thanks.

**Comment by rarkins** [comment]
Try this:

```
  "regexManagers": [
    {
      "fileMatch": [".*molecule.yml$"],
      "matchStrings": [
        "image: (?<depName>.*?):(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)"
      ],
      "datasourceTemplate": "docker",
      "commitMessagePrefix": "test(renovate):"
    }
  ],
  "regex": {
	"commitMessagePrefix": "test(renovate):"
  }
```

Alternatively, put it in `packageRules` using `"managers": ["regex"]`.

**Comment by micheelengronne** [comment]
Thank you. What can I do if I have different regex rules that need different commitMessagePrefix ?

**Comment by micheelengronne** [comment]
If I do that :

```
  "docker": {
    "commitMessagePrefix": "test(renovate):"
  },
  "regexManagers": [
    {
      "fileMatch": [".*molecule.yml$"],
      "matchStrings": [
        "image: (?<depName>.*?):(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)"
      ],
      "datasourceTemplate": "docker"
    }
  ]
```

Does it work ? As the datasourceTemplate is docker.

**Comment by rarkins** [comment]
Let's test again after https://github.com/renovatebot/renovate/pull/6024 is merged. I think the current code we have is causing problems with regex docker managers right now.

**Comment by rarkins** [comment]
Why did you want to put it inside `docker` instead of `regex` though? Right now the `docker {}` syntax is short for docker manager, not docker datasource

**Comment by micheelengronne** [comment]
Ok, I misunderstood that. I wanted to ensure that only regex corresponding to docker datasource had this commitMessagePrefix and other regex had not.

If I do smthg like that:

```
  "regexManagers": [
    {
      "fileMatch": [".*molecule.yml$"],
      "matchStrings": [
        "image: (?<depName>.*?):(?<currentValue>.*?)@(?<currentDigest>sha256:[a-f0-9]+)"
      ],
      "datasourceTemplate": "docker"
    }
  ],
  "packageRules": [
    {
      "paths": [".*molecule.yml$"],
      "extends": [":semanticCommitTypeAll(chore)"]
    }
  ]
```

Does it work ?

**Comment by rarkins** [comment]
Yes, your config looks to match your description. I think you'll still get an error at runtime but that hopefully will be resolved once the commit from #6024 is released. If you're using the app, it will take at least a few hours to put live though.

**Comment by micheelengronne** [comment]
ok thanks.
